src/DiffEqGPU.jl

```diff
@@ -3,7 +3,48 @@ module DiffEqGPU
 using GPUifyLoops, CuArrays, CUDAnative, DiffEqBase
 function gpu_kernel(f,du,u,p,t)
     @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
-        @views @inbounds f(du[:,i],u[:,i],p,t)
+        @views @inbounds f(du[:,i],u[:,i],p[:,i],t)
+        nothing
+    end
+    nothing
+end
+
+function jac_kernel(f,J,u,p,t)
+    @loop for i in (0:(size(u,2)-1); (blockIdx().x-1) * blockDim().x + threadIdx().x)
+        section = 1 + (i*size(u,1)) : ((i+1)*size(u,1))
+        @views @inbounds f(J[section,section],u[:,i+1],p[:,i+1],t)
+        nothing
+    end
+    nothing
+end
+
+function discrete_condition_kernel(condition,cur,u,t,p)
+    @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
+        @views @inbounds cur[i] = condition(u[:,i],t,FakeIntegrator(u[:,i],t,p[:,i]))
+        nothing
+    end
+    nothing
+end
+
+function discrete_affect!_kernel(affect!,cur,u,t,p)
+    @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
+        @views @inbounds cur[i] && affect!(FakeIntegrator(u[:,i],t,p[:,i]))
+        nothing
+    end
+    nothing
+end
+
+function continuous_condition_kernel(condition,out,u,t,p)
+    @loop for i in (1:size(u,2); (blockIdx().x-1) * blockDim().x + threadIdx().x)
+        @views @inbounds out[i] = condition(u[:,i],t,FakeIntegrator(u[:,i],t,p[:,i]))
+        nothing
+    end
+    nothing
+end
+
+function continuous_affect!_kernel(affect!,event_idx,u,t,p)
+    @loop for i in ((event_idx,); (blockIdx().x-1) * blockDim().x + threadIdx().x)
```

Comment:
@vchuravy I want to do the `affect!` in one kernel call, but it only needs to be done at `event_idx`. Just calling that `affect!` would give multiple kernel calls (with scalar indexing) because the user will use `u` directly, so does this make sense?

Reply:
Sweet, Valentin approved hack.

Reply:
Seems fine :)